### PR TITLE
Decouple user credentials for external data sources from datasets

### PR DESCRIPTION
**Pull request by lebinh**
For PE-1167

This go together with https://github.com/ddf-project/DDF/pull/255

JdbcDDFManager can be created without a JDBCDataSourceDescriptor but using a source URI instead. Also added the createDDF method to JdbcDDFManager.

### CODE REVIEW

**Comment by lebinh** [comment]
@PangZhi pls help review

File: postgres/src/main/scala/io/ddf/postgres/PostgresDDFManager.scala [hunk 1]
```diff
@@ -13,7 +13,19 @@ import io.ddf.jdbc.content.{Catalog, SqlArrayResultCommand}
 import io.ddf.jdbc.utils.Utils
 import scalikejdbc.{DB, SQL}
 
-class PostgresDDFManager(dataSourceDescriptor: DataSourceDescriptor, engineType: EngineType) extends JdbcDDFManager(dataSourceDescriptor, engineType) {
+class PostgresDDFManager(dataSourceDescriptor: DataSourceDescriptor,
```
Comment (nhanitvn) [inline postgres/src/main/scala/io/ddf/postgres/PostgresDDFManager.scala]: why a blank line here? :)

Comment (lebinh) [inline postgres/src/main/scala/io/ddf/postgres/PostgresDDFManager.scala]: I break it to stay within 120 chars limit

**Comment by lebinh** [comment]
retest this please

**Comment by hai-adatao** [comment]
retest this please

**Pull request merged**